### PR TITLE
[dpdk] Update to version v22.03

### PR DESCRIPTION
**Pull request by ljishen**
**Describe the pull request**

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  I've removed the `x64-linux=fail` from the file to mark that this port is supported by that Linux triplet.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  `Yes`

The reason to update the numactl port here is that the build process requires the presence of both the static and dynamic libraries of numactl. Otherwise, the exact same error reported here will appear (https://mails.dpdk.org/archives/dev/2019-May/132290.html).

### CODE REVIEW

**Comment by LilyWangLL** [comment]
> The reason to update the numactl port here is that the build process requires the presence of both the static and dynamic libraries of numactl. Otherwise, the exact same error reported here will appear (https://mails.dpdk.org/archives/dev/2019-May/132290.html).

This link open failed on my local, is there somewhere else that explains this?

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -203,14 +203,13 @@ dlib:arm-uwp=fail
 dlib:x64-uwp=fail
 dmlc:arm-uwp=fail
 dmlc:x64-uwp=fail
-dpdk:arm-uwp=fail
```
Comment (LilyWangLL) [inline scripts/ci.baseline.txt:206]: If here only remove `dpdk:x64-linux=fail`, please don't change other line.

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by ljishen** [comment]
Is that because of the Cloudflare worldwide down issue? Can you try the link again? I can open it here.

File: scripts/ci.baseline.txt (hunk 1 above)
Comment (ljishen) [reply scripts/ci.baseline.txt:206]: OK, I will update the PR soon.

**Comment by LilyWangLL** [comment]
> Is that because of the Cloudflare worldwide down issue? Can you try the link again? I can open it here.

I can open it now, thanks~

File: ports/dpdk/portfile.cmake [hunk 2]
```diff
@@ -1,43 +1,94 @@
...
-FILE(INSTALL ${SOURCE_PATH}/Debug/lib/ DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib)
-FILE(INSTALL ${REAL_FILES} DESTINATION ${CURRENT_PACKAGES_DIR}/include/${PORT})
-FILE(INSTALL ${CMAKE_CURRENT_LIST_DIR}/dpdkConfig.cmake DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
-FILE(INSTALL ${CMAKE_CURRENT_LIST_DIR}/usage DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
-FILE(INSTALL ${SOURCE_PATH}/license/README DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+if(NOT VCPKG_CMAKE_SYSTEM_NAME
+   OR NOT (VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux" OR VCPKG_CMAKE_SYSTEM_NAME
+                                                       STREQUAL "FreeBSD"))
+  message(
+    FATAL_ERROR "Intel DPDK currently only supports Linux/FreeBSD platforms")
```
Comment (LilyWangLL) [inline ports/dpdk/portfile.cmake:5]:
```suggestion
if(NOT VCPKG_CMAKE_SYSTEM_NAME OR NOT (VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux" OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "FreeBSD"))
  message(FATAL_ERROR "Intel DPDK currently only supports Linux/FreeBSD platforms")
```

Comment (LilyWangLL) [reply ports/dpdk/portfile.cmake:5]: It looks weird😅

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by dg0yt** [comment]
> The reason to update the numactl port here is that the build process requires the presence of both the static and dynamic libraries of numactl. Otherwise, the exact same error reported here will appear (https://mails.dpdk.org/archives/dev/2019-May/132290.html).

Hm, I don't think that dpdk's difficulties to separate static from dynamic linkage should be proliferated to other ports.

**Comment by ljishen** [comment]
@dg0yt OK. I'll separate the PR into two each for a different port.

**Comment by ljishen** [comment]
@dg0yt @LilyWangLL I've closed this one and replaced it with two separate PRs #25372 and #25374.

**Comment by dg0yt** [comment]
@ljishen There was no need to change to new PRs. This just hides the existing review comments. You could have added reverts to, or even force-pushed, this PR's branch.

And to be clear, the change to numactl is invalid IMO. You need to find the right fix for the problem from https://mails.dpdk.org/archives/dev/2019-May/132290.html.

**Pull request closed without merging**